Logs Sample Data

alexfrancoeur opened the pull request:
This is one of the data sets we'd like to add for sample data. This is my first take on the data set itself so open to any feedback.

We're using makelogs as the basis here but want to tweak the URL's and requests to be more Elastic-y.

Open to feedback on the dataset, dashboards, etc.

Also note that I did not add any tests. There were a bunch of sample flight data, I'm not sure if they're necessary for other sample data sets. I'll look for @nreese here for guidance and can add some in if necessary. 

### Screenshots

<img width="1675" alt="screen shot 2018-08-22 at 5 25 40 pm" src="https://user-images.githubusercontent.com/25181823/44491867-8a13e200-a630-11e8-9ec2-35a4acfc04dc.png">

<img width="1677" alt="screen shot 2018-08-22 at 5 26 03 pm" src="https://user-images.githubusercontent.com/25181823/44491872-8ed89600-a630-11e8-9cdd-afc2c72c20c4.png">

![image](https://user-images.githubusercontent.com/25181823/44732945-76470080-aab4-11e8-967a-88777fd745ab.png)

### Sample document

``` json
{
  "_index": "kibana_sample_data_logs",
  "_type": "_doc",
  "_id": "hikLbGUBoflyAT6o14Fv",
  "_version": 1,
  "_score": null,
  "_source": {
    "referer": "http://www.elastic-elastic-elastic.com/success/kenneth-ham",
    "clientip": "109.157.83.197",
    "response": "200",
    "tags": [
      "success",
      "info"
    ],
    "message": "109.157.83.197 - - [2018-08-03T13:07:45.046Z] \"GET /apm-server/apm-server-6.3.2-windows-x86.zip HTTP/1.1\" 200 5782 \"-\" \"Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1\"",
    "request": "/apm-server/apm-server-6.3.2-windows-x86.zip",
    "geo": {
      "srcdest": "IN:AO",
      "src": "IN",
      "dest": "AO",
      "coordinates": {
        "lat": 39.21837556,
        "lon": -75.59642667
      }
    },
    "phpmemory": null,
    "extension": "zip",
    "index": "kibana_sample_data_logs",
    "url": "https://artifacts.elastic.co/downloads/apm-server/apm-server-6.3.2-windows-x86.zip",
    "memory": null,
    "host": "artifacts.elastic.co",
    "utc_time": "2018-08-03T13:07:45.046Z",
    "machine": {
      "ram": 8589934592,
      "os": "win 7"
    },
    "agent": "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1",
    "ip": "109.157.83.197",
    "bytes": 5782,
    "timestamp": "2018-08-24T13:07:45.046Z"
  },
  "fields": {
    "utc_time": [
      "2018-08-03T13:07:45.046Z"
    ],
    "hour_of_day": [
      13
    ],
    "timestamp": [
      "2018-08-24T13:07:45.046Z"
    ]
  },
  "sort": [
    1535116065046
  ]
}
```

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/696/)

alexfrancoeur commented:
Updated PR so that it now has a working index pattern. New screenshot and sample doc above.

@jamiesmith @asawariS would love some feedback on the visualization types shown here, metrics to track, etc. Any thoughts on a color schema would be great as well. I'd still like to add annotations to the TSVB chart and better metrics in the table.  There needs to be some cleanup, but this is actually a dashboard I use for demo's pretty frequently. More to show off functionality, not necessarily tell a story. 

@jamiesmith I know we talked about adding "Elasticy" hosts, requests, etc. Any input here from PMM would be great, I'd really like to make this data fun. I know that you were also thinking of anomalies to add in. Note that I made a few changes in the Sample Data Clean Up doc and removed `@` from a few of the field names. I also updated the index to be `kibana_sample_data_logs`. 

@nreese open question here around tests. Do I need to add them in if there are already some for flight sample data? I'm guessing I do, but thought I'd ask before I write any 😄

nreese commented:
> open question here around tests. Do I need to add them in if there are already some for flight sample data? I'm guessing I do, but thought I'd ask before I write any

I think we should add a test that installs the sample data set, opens the dashboard and makes sure the expected number of panels are opened, and then uninstalls the sample data set. I can work on this and give you a PR with the tests.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/728/)

alexfrancoeur commented:
@nreese I should be able to add them, will ping with questions. Thanks for the input!

alexfrancoeur commented:
@nreese I'm assuming I can pretty much copy what you did with flights here

nreese commented:
@alexfrancoeur yes, just don't copy the skipped tests. Simply opening the dashboard and verifying the panel count is good enough for now.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/757/)

alexfrancoeur commented:
Still iterating on the data set a bit with @jamiesmith, updated screenshot and doc in the summary. We're close!

alexfrancoeur commented:
@nreese how do those tests look?

nreese commented:
@alexfrancoeur tests look great. Nice work adding another data set

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/864/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/969/)

alexfrancoeur commented:
Cleaned up the dashboard with help from @asawariS 

![image](https://user-images.githubusercontent.com/25181823/44667614-4121ac00-a9e9-11e8-8164-531017bec155.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/984/)

alexfrancoeur commented:
@jimgoodwin @rayafratkina @epixa We have buy in from PMM and @nreese has reviewed the code. Assuming we are green, any reasons for not merging?

epixa commented:
It just needs a proper LGTM

alexfrancoeur commented:
@nreese can I get a LGTM?

nreese commented:
Can you re-sync with master. I am not seeing the changes from https://github.com/elastic/kibana/pull/21655/ in your branch?

alexfrancoeur commented:
@nreese any better? should be up to date now.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1106/)

nreese commented:
@alexfrancoeur 

Could you remove the checks for toast notifications from you functional tests. Those are flaky because loading the sample data calls `await sampleDataCard.waitForDeletedByClassName('euiLoadingSpinner');` and this call could take longer than the toast notification is displayed.

```
fail: "homepage app sample data should install logs sample data set"
16:30:28            │      Error: expected false to equal true
```

nreese commented on test/functional/apps/home/_sample_data.js:
```diff
@@ -87,15 +106,40 @@ export default function ({ getService, getPageObjects }) {
       });
+
+      it('should launch sample logs data set dashboard', async ()=> {
+        await PageObjects.home.launchSampleDataSet('logs');
```
I could not get this test to pass locally. At this point, the browser is on the `flights` dashboard so it can not find `logs` sample data card. I think you should just be able to change the `after` block (line 67) to `afterEach`

alexfrancoeur replied:
@nreese I _think_ I should be good now, modified to `afterEach`

alexfrancoeur replied:
ah, I know why it's failing

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1113/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1127/)

nreese commented on test/functional/apps/home/_sample_data.js:
```diff
@@ -87,15 +106,40 @@ export default function ({ getService, getPageObjects }) {
         expect(tsvb).to.be(true);
       });
+
+      it.skip('should launch sample logs data set dashboard', async ()=> {
```
This test should not be skipped. It is failing because `PageObjects.home.launchSampleDataSet('logs')` needs to be run from from the kibana home page `tutorial_directory/sampleData`. You had it right by changing `after` to `afterEach`.

alexfrancoeur replied:
:doh:

nreese commented on test/functional/apps/home/_sample_data.js:
```diff
@@ -87,15 +106,40 @@ export default function ({ getService, getPageObjects }) {
+      it('should uninstall logs sample data set', async ()=> {
+        await PageObjects.home.removeSampleDataSet('logs');
+        await retry.try(async () => {
+          const successToastExists = await PageObjects.home.doesSampleDataSetSuccessfulUninstallToastExist();
```
Remove the check for the toast notifications. Those are flaky.

alexfrancoeur replied:
++ removed, testing locally (because I just figured out how to do that 😄 )

nreese commented on test/functional/apps/home/_sample_data.js:
```diff
@@ -32,26 +32,44 @@ export default function ({ getService, getPageObjects }) {
+    it('should install logs sample data set', async ()=> {
+      await PageObjects.home.addSampleDataSet('logs');
+      await retry.try(async () => {
+        const successToastExists = await PageObjects.home.doesSampleDataSetSuccessfulInstallToastExist();
```
Remove the check for the toast notifications. Those are flaky.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1137/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1149/)

AlonaNadler commented:
@alexfrancoeur  Great job on the sample web logs!!  
Few suggestions:
* Can you add trend logs to the TSVB table, it's a features users don't know exists, even if it won't make sense it demonstrate a capability we have 
![image](https://user-images.githubusercontent.com/32550316/44759200-76afbd80-aaed-11e8-95c8-76cb2a983438.png)
- Why do you have `-` in the custom label, it looks like there is an error when it's presented in the dashboard
<img width="367" alt="screen shot 2018-08-28 at 18 09 38" src="https://user-images.githubusercontent.com/32550316/44759375-a3b0a000-aaee-11e8-8f33-02ef1ed2b3c7.png">

- I prefer to have panels titles, users are dropped to this dashboard, any additional context can help, especially when it comes to logs when your metrics might inspire new ideas in them
- Change the prefix you give to all objects from `[logs]` to `[sample logs]` since it might get mixed with any future log users will ingest
- Change the title of `Sample log data` to `Sample web logs` these 2 panels looks identical, it will be easier to differentiate when just quickly scanning them 
![image](https://user-images.githubusercontent.com/32550316/44759345-81b71d80-aaee-11e8-9817-eccde22a53ad.png)

nreese commented:
> @alexfrancoeur @nreese Great job on the sample web logs!!

It was all @alexfrancoeur.

alexfrancoeur commented:
@nreese stop being modest - I copied all your code 😄 

@AlonaNadler great feedback! Updated and committed. Once green, I'll merge

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1221/)

nreese commented:
jenkins, test this

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1269/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1338/)
